src/flag-evaluation-details.ts

```diff
@@ -0,0 +1,127 @@
+import { Allocation, Variation } from './interfaces';
+import { Rule } from './rules';
+
+export const flagEvaluationCodes = [
+  'MATCH',
+  'FLAG_UNRECOGNIZED_OR_DISABLED',
+  'TYPE_MISMATCH',
+  'ASSIGNMENT_ERROR',
+  'DEFAULT_ALLOCATION_NULL',
+] as const;
+
+export type FlagEvaluationCode = typeof flagEvaluationCodes[number];
+
+export enum AllocationEvaluationCode {
+  UNEVALUATED = 'UNEVALUATED',
+  MATCH = 'MATCH',
+  BEFORE_START_TIME = 'BEFORE_START_TIME',
+  TRAFFIC_EXPOSURE_MISS = 'TRAFFIC_EXPOSURE_MISS',
+  AFTER_END_TIME = 'AFTER_END_TIME',
+  FAILING_RULE = 'FAILING_RULE',
+}
+
+export interface AllocationEvaluation {
+  key: string;
+  allocationEvaluationCode: AllocationEvaluationCode;
+  orderPosition: number;
+}
+
+export interface FlagEvaluationDetails {
+  variationKey: string | null;
+  variationValue: Variation['value'] | null;
+  flagEvaluationCode: FlagEvaluationCode;
+  flagEvaluationDescription: string;
+  matchedRule: Rule | null;
+  matchedAllocation: AllocationEvaluation | null;
+  unmatchedAllocations: Array<AllocationEvaluation>;
+  unevaluatedAllocations: Array<AllocationEvaluation>;
+}
+
+export class FlagEvaluationDetailsBuilder {
+  private variationKey: FlagEvaluationDetails['variationKey'];
+  private variationValue: FlagEvaluationDetails['variationValue'];
+  private matchedRule: FlagEvaluationDetails['matchedRule'];
+  private matchedAllocation: FlagEvaluationDetails['matchedAllocation'];
+  private unmatchedAllocations: FlagEvaluationDetails['unmatchedAllocations'];
+  private unevaluatedAllocations: FlagEvaluationDetails['unevaluatedAllocations'];
+
+  constructor(private readonly allocations: Allocation[]) {
+    this.setNone();
+  }
+
+  setNone = (): FlagEvaluationDetailsBuilder => {
+    this.variationKey = null;
+    this.variationValue = null;
+    this.matchedAllocation = null;
+    this.matchedRule = null;
+    this.unmatchedAllocations = [];
+    this.unevaluatedAllocations = [];
+    return this;
+  };
+
+  setNoMatchFound = (
+    unmatchedAllocations: Array<AllocationEvaluation> = [],
+  ): FlagEvaluationDetailsBuilder => {
+    this.variationKey = null;
+    this.variationValue = null;
+    this.matchedAllocation = null;
+    this.matchedRule = null;
+    this.unmatchedAllocations = unmatchedAllocations;
+    this.unevaluatedAllocations = this.allocations.map(
+      (allocation, i) =>
+        ({
+          key: allocation.key,
+          allocationEvaluationCode: AllocationEvaluationCode.UNEVALUATED,
+          orderPosition: i,
+        } as AllocationEvaluation),
+    );
```

Comment:
Whoops this is backwards. If no match was found, then all allocations were evaluated.

Reply:
oh yeah you'll want to fix that

Reply:
Fixed in https://github.com/Eppo-exp/js-client-sdk-common/pull/100